plenum/test/node_catchup_with_3pc/test_catchup_with_skipped_commits.py

```diff
@@ -55,29 +42,53 @@ def check_lagging_node_is_not_syncing_audit():
     def check_lagging_node_done_catchup():
         assert lagging_node_state() == NodeLeecherService.State.Idle
 
+    def check_nodes_ordered_till(nodes: Iterable, view_no: int, pp_seq_no: int):
+        for node in nodes:
+            assert compare_3PC_keys((view_no, pp_seq_no), node.master_replica.last_ordered_3pc) >= 0
+
     # Preload nodes with some transactions
     sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
     for node in txnPoolNodeSet:
         assert node.master_replica.last_ordered_3pc == (0, 1)
 
-    # Delay some commits on lagging node
-    some_commits = start_delaying(lagging_stasher, delay_some_commits(0, [2, 3]))
+    # Setup delayers
+    lagging_mid_commits = start_delaying(lagging_stasher, delay_3pc(after=3, before=6, msgs=Commit))
```

Comment:
- delay Commits for ppSeqNo=4 on NodeX
- increase the timeout for out of order commits on NodeX to 10000 secs
- send 5 batches
- make sure that NodeX ordered 3 of them
- make sure that others ordered ppSeqNo=5
- make sure that NodeX has out of order commits for ppSeqNo=5
- process delayed commits for view4 on NodeX
- make sure that NodeX ordered till batch ppSeqNo=4
- delay POOL catchup on NodeX
- start catchup on Node X
- make sure that audit catchup is done, and domain one is started
- call processOutOfOrderCommits by hand (emulating call by timeout)   
    - it should not do anything on the latest master
    - it will order ppSeqNo=5 before starting Domain catchup on old code
- finish catchup (process delayed catcup)
=> ppSeqNo=5 is caught up, all nodes have equal data, and NodeX can participate in more ordering
    - on old code, NodeX will have duplicate txns for ppSeqNo=5: one from catchup, and one from outOfOrderCommits
- 

- start catchup of NodeX